sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -581,33 +581,32 @@ def test_multinomial():
     assert multinomial(1, 1, 1) == 6
     assert multinomial(S.Half, S.Half).equals(4/pi)
 
-    assert unchanged(multinomial, k, n + k)
+    assert multinomial(k, n + k) == binomial(n + 2*k, k)
     assert multinomial(k, n + k).rewrite(binomial
-        ) == binomial(n + 2*k, n + k)
+        ) == binomial(n + 2*k, k)
     # don't rewrite if an arg sign (e.g. sign(n - k)) could be negative
     m = multinomial(k, n - k)
     assert m.rewrite(binomial) == m
 
     assert multinomial(n, n).is_integer
     assert multinomial(nt, k).is_integer is False
 
-    assert expand_func(multinomial(n, -2)) == 0
-    assert (expand_func(multinomial(n, -pi))).equals(
-        (pi - n)*gamma(n - pi)/(pi*n*gamma(-pi)*gamma(n)))
+    assert multinomial(n, -2) == 0
+    assert multinomial(n, -pi).rewrite(gamma
+        ) == gamma(n - pi + 1)/(gamma(1 - pi)*gamma(n + 1))
     assert expand_func(multinomial(n, 2)) == (n + 1)*(n + 2)/2
     assert expand_func(multinomial(n, 2, 3)) == (n + 1)*(
         n + 2)*(n + 3)*(n + 4)*(n + 5)/12
-    assert expand_func(multinomial(n, n + 3)
-        ) == binomial(2*n + 3, n + 3)
+    assert multinomial(n, n + 3) == binomial(2*n + 3, n)
 
     assert multinomial(n, 3).expand(func=True) ==  n**3/6 + n**2 + 11*n/6 + 1
 
-    assert multinomial(kp, kp + 1).expand(func=True
-        ) == binomial(2*kp + 1, kp + 1)
+    assert multinomial(kp, kp + 1) == binomial(2*kp + 1, kp)
 
     assert multinomial(n, z) == 1
     assert multinomial(z) == 1
-    assert unchanged(multinomial, z, n, k)  # TODO remove zeros
+    assert multinomial(z, n, k) == binomial(k + n, k)
+    assert unchanged(multinomial, z, z, n, k, k)  # TODO remove zeros; order args
```

Comment:
We need and `__init__` or `__new__` or something so this will come back as `multinomial(k, k, n)`

Reply:
new works best, are we only removing symbolic zeros or even numbers @smichr,
cause if we remove numbers ... then multinomial(0, 0) is a problem

Reply:
This (0,0) case is handled by re-inserting 0s for any arg list whose length reduces to less than 2. If you can write a `__new__` routine that does something like the following, that would be good.
```python
if not args:
    raise error
args = [i for i in ordered(args) if not i.is_zero]
if len(args) < 2:
    args = args + [S.Zero]*(2 - len(args))
```

Reply:
Ohhh okay!
the one I wrote so far works like this
```python3
>>> multinomial(z,z,n,k,k)
multinomial(n, k, k)
>>> multinomial(z)
1
>>> multinomial(0)
1
>>> multinomial(z,z,n,k)
binomial(k + n, k)
>>> 
```

Reply:
but I didn't consider the full picture 😅 ... turns out binomial(z,z) which should be equal to 1, gives ValueError cause z gets removed

Reply:
Current result is
```python
In [1]: multinomial(*[var('z',zero=True)]*2)
Out[1]: 1
```